src/api/c/internal.cpp

```diff
@@ -0,0 +1,179 @@
+/*******************************************************
+ * Copyright (c) 2016, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/dim4.hpp>
+#include <af/device.h>
+#include <af/version.h>
+#include <af/internal.h>
+#include <backend.hpp>
+#include <platform.hpp>
+#include <Array.hpp>
+#include <handle.hpp>
+#include "err_common.hpp"
+#include <cstring>
+
+using namespace detail;
+
+af_err af_create_array_with_strides(af_array *arr,
+                                    const void *data,
+                                    const dim_t offset,
+                                    const unsigned ndims,
+                                    const dim_t *const dims_,
+                                    const dim_t *const strides_,
+                                    const af_dtype ty,
+                                    const af_source location)
+{
+    try {
+        ARG_ASSERT(2, offset >= 0);
```

Comment:
You'll need to add af_init here as it a creation function. Like https://github.com/arrayfire/arrayfire/blob/devel/src/api/c/data.cpp#L78